vendor/bat-native-ads/src/bat/ads/internal/processors/behavioral/bandits/epsilon_greedy_bandit_processor.cc

```diff
@@ -25,13 +26,13 @@ namespace {
 constexpr double kDefaultArmValue = 1.0;
 constexpr int kDefaultArmPulls = 0;
 
-targeting::EpsilonGreedyBanditArmMap MaybeAddOrResetArms(
-    const targeting::EpsilonGreedyBanditArmMap& arms) {
-  targeting::EpsilonGreedyBanditArmMap updated_arms = arms;
+void MaybeAddOrResetArms(targeting::EpsilonGreedyBanditArmMap* arms) {
+  DCHECK(arms);
 
-  for (const char* const segment : targeting::GetSegments()) {
-    const auto iter = updated_arms.find(segment);
-    if (iter != updated_arms.cend()) {
+  for (const base::StringPiece value : targeting::GetSegments()) {
+    const std::string segment = static_cast<std::string>(value);
+    const auto iter = arms->find(segment);
```

Comment:
why not `value.data()`? constructing `std::string` on each cycle is not great.

Reply:
If pass `value.data()` then `std::string` will be constructed under the hood. But also it will be constructed during `insert_or_assign` call. So `std::string segment = static_cast<std::string>(value); ` is required to avoid double construction.

Reply:
I see. I thought `find()` can do a comparison without constructing `std::string` through some template magic, but testing in godbolt showed that the `std::string` construction is indeed required.